ios/RNVIClientModule.m

```diff
@@ -291,23 +283,21 @@ - (void)stopObserving {
 }
 
 RCT_REMAP_METHOD(createAndStartCall,
-                  callUser:(NSString *)user
-                  withVideoSettings:(NSDictionary *)videoFlags
-                  withVideoCodec:(VIVideoCodec)videoCodec
-                  customData:(NSString *)customData
-                  headers:(NSDictionary *)headers
-                  setupCallKit:(BOOL)setupCallKit
+                 callUser:(NSString *)user
+                 withSettings:(NSDictionary *)settings
                  responseCallback:(RCTResponseSenderBlock)callback) {
     if (_client) {
         VICallSettings *callSettings = [[VICallSettings alloc] init];
-        callSettings.customData = customData;
-        callSettings.extraHeaders = headers;
+        callSettings.customData = [settings valueForKey:@"customData"];
+        callSettings.extraHeaders = [settings valueForKey:@"extraHeaders"];;
+        NSDictionary *videoFlags = [settings valueForKey:@"video"];
         callSettings.videoFlags = [VIVideoFlags videoFlagsWithReceiveVideo:[[videoFlags valueForKey:@"receiveVideo"] boolValue]
                                                                  sendVideo:[[videoFlags valueForKey:@"sendVideo"] boolValue]];
-        callSettings.preferredVideoCodec = videoCodec;
+        callSettings.preferredVideoCodec = [self convertVideoCodecFromString:[settings valueForKey:@"preferredVideoCodec"]];
+        callSettings.enableSimulcast = [settings valueForKey:@"enableSimulcast"];
```

Comment:
this piece of code can be moved to a private method and reused for `callConference` and `answerCall`